MovingBlocks/Terasology
# Move controller settings to new page, make ControllerInfo generic

### Contains

Fixes #3648. I moved the controller settings to a separate UI page. The page lists each connected controller and its settings. To do this I fleshed out ```ControllerInfo``` with lists of buttons, axes, and hats, so that the info is more generic and conforms more to the information that ```JInput``` provides. This also required some changes in ```InputSystem``` and ```JInputControllerDevice``` for handling the new axes (there are no changes related to handling of buttons, I wanted to make minimal possible changes).

### How to test

Connect a controller. Go to Settings -> Controllers and you will see all connected controllers.
Some notes about testing (possibly for future issues):
- The system seems to assume only one connected controller. It seems that ```AbstractBindableAxis``` has no reference to a particular controller, so multiple controllers would probably be able to control the one binding (I didn't test this). Same with ```BindableButton```. Because of this I've listed all button bindings at the bottom of the settings page.
- Axis inversion appears to be buggy. I can see from logging that ```InputSystem.processControllerAxisInput``` is inverting ```axis.setTargetValue```, but the resultant movement in game still goes in the non-inverted direction. I was also able to re-create on the ```develop``` branch (n.b. I only looked at this very briefly).
